src/test/java/org/verapdf/wcag/algorithms/semanticalgorithms/ContrastRatioCheckerTests.java

```diff
@@ -0,0 +1,163 @@
+            StringBuilder stb = new StringBuilder();
+            stb.append(entry.getValue()).append("  - ").append(entry.getTotalOccurrence()).append(" :   ");
+            int relativePointCount = entry.getTotalOccurrence() * 1500 / absoluteMax / 10;
+            for (int i = 0; i < relativePointCount; i++) {
+                stb.append('*');
+            }
+            result.add(stb.toString());
+        }
+        return result;
+    }
+
+    private List<String> presentMapAsPercentHistogram(List<ContrastRatioChecker.DataPoint> source) {
+        List<String> result = new ArrayList<>();
+        int absoluteMax = 0;
+        for (ContrastRatioChecker.DataPoint entry : source) {
+            if (entry.getTotalOccurrence() > absoluteMax) {
+                absoluteMax = entry.getTotalOccurrence();
+            }
+        }
+        int[] percentArray = new int[101];
+        for (ContrastRatioChecker.DataPoint entry : source) {
+            percentArray[(int)Math.round(entry.getValue() * 100)] += entry.getTotalOccurrence();
+        }
+
+        for(int j = 0; j < 101; j++) {
+            percentArray[j] = percentArray[j] * 1000 / absoluteMax / 10;
+        }
+
+        System.out.println("Percent : value 'histogram'");
+        for (int j = 0; j < 101; j++) {
+            StringBuilder stb = new StringBuilder();
+            stb.append(j).append('-').append(j + 1).append( " : ");
+            for (int k = 0; k <= percentArray[j]; k++) {
+                stb.append('*');
+            }
+            result.add(stb.toString());
+        }
+        return result;
+    }
+
+    private int[] presentListAsHistogramData(List<ContrastRatioChecker.DataPoint> source) {
+        int absoluteMax = 0;
+        for (ContrastRatioChecker.DataPoint entry : source) {
+            if (entry.getTotalOccurrence() > absoluteMax) {
+                absoluteMax = entry.getTotalOccurrence();
+            }
+        }
+        int [] percentArray = new int[101];
+        for (ContrastRatioChecker.DataPoint entry : source) {
+            percentArray[(int)Math.floor(entry.getValue() * 100)] += entry.getTotalOccurrence();
+        }
+        return percentArray;
+    }
+
+    private void printHistogramToFile(int [] dataArray, String filePath) {
+        DefaultCategoryDataset dataset = new DefaultCategoryDataset();
+
+        for (int i = 0; i < dataArray.length; i++) {
+            dataset.addValue(dataArray[i], "", String.valueOf(i));
+        }
+        String plotTitle = "Relative luminance";
+        String xaxis = "luminance percent";
+        String yaxis = "pixelsPresent";
+        PlotOrientation orientation = PlotOrientation.VERTICAL;
+        boolean showLegend = false;
+        boolean toolTips = false;
+        boolean urls = false;
+        JFreeChart chart = ChartFactory.createBarChart( plotTitle, xaxis, yaxis,
+                dataset, orientation, showLegend, toolTips, urls);
+        int width = 2500;
+        int height = 1500;
+        try {
```

Comment:
Useless try {} catch{}, just throw exception from this method

Reply:
ok, done